From 76899a7a6dbdee5b85a2cc17942e9fef9abdf73e Mon Sep 17 00:00:00 2001
From: bashonly <88596187+bashonly@users.noreply.github.com>
Date: Sat, 21 Oct 2023 23:51:55 +0300
Subject: [PATCH 2/2] Revert "fftools/ffmpeg_mux_init: drop a duplicated block
 in copy_metadata()"

This reverts commit 3c7dd5ed37da6d2de06c4850de5a319ca9cdd47f.
---
 fftools/ffmpeg_mux_init.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/fftools/ffmpeg_mux_init.c b/fftools/ffmpeg_mux_init.c
index 4cf18b4f02..4664ad9baa 100644
--- a/fftools/ffmpeg_mux_init.c
+++ b/fftools/ffmpeg_mux_init.c
@@ -2182,6 +2182,16 @@ static int copy_metadata(Muxer *mux, AVFormatContext *ic,
     if (ret < 0)
         return ret;
 
+    if (!ic) {
+        if (type_out == 'g' || !*outspec)
+            *metadata_global_manual = 1;
+        if (type_out == 's' || !*outspec)
+            *metadata_streams_manual = 1;
+        if (type_out == 'c' || !*outspec)
+            *metadata_chapters_manual = 1;
+        return 0;
+    }
+
     if (type_in == 'g' || type_out == 'g')
         *metadata_global_manual = 1;
     if (type_in == 's' || type_out == 's')
-- 
2.39.2

